tests/test_http.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest.mock import patch, MagicMock
+
+from aioapns.common import NotificationResult
+
+from tests import testutils
+
+PUSHKIN_ID_1 = "com.example.apns"
+PUSHKIN_ID_2 = "*.example.*"
+PUSHKIN_ID_3 = "com.example.a*"
+
+TEST_CERTFILE_PATH = "/path/to/my/certfile.pem"
+
+# Specific app id
+DEVICE_EXAMPLE_SPECIFIC = {
+    "app_id": "com.example.apns",
+    "pushkey": "spqr",
+    "pushkey_ts": 42,
+}
+
+# Only one time matching app id (with PUSHKIN_ID_2)
+DEVICE_EXAMPLE_MATCHING = {
+    "app_id": "com.example.bpns",
+    "pushkey": "spqr",
+    "pushkey_ts": 42,
+}
+
+# More than one times matching app id (with PUSHKIN_ID_2 and PUSHKIN_ID_3)
+DEVICE_EXAMPLE_AMBIGIOUS = {
+    "app_id": "com.example.apns2",
+    "pushkey": "spqr",
+    "pushkey_ts": 42,
+}
+
+
+class HttpTestCase(testutils.TestCase):
+    def setUp(self):
+        self.apns_mock_class = patch("sygnal.apnspushkin.APNs").start()
```

Comment:
Given `APNs` is only present in `sygnal.apnspushkin` as an import from `aioapns`, why not use `aioapns.APNs` here?

Reply:
Hmm, it wasn't a conscious choice. I just copied it from another test class. Updated.

Reply:
Well, i could not make it work with `aioapns`. Tests were failed. Had to revert.